expression/builtin_time.go

```diff
@@ -401,7 +401,16 @@ func (c *timeDiffFunctionClass) getFunction(ctx sessionctx.Context, args []Expre
 	arg0FieldTp, arg1FieldTp := args[0].GetType(), args[1].GetType()
 	arg0Tp, arg1Tp := c.getArgEvalTp(arg0FieldTp), c.getArgEvalTp(arg1FieldTp)
 	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETDuration, arg0Tp, arg1Tp)
-	bf.tp.Decimal = mathutil.Max(arg0FieldTp.Decimal, arg1FieldTp.Decimal)
+
+	arg0Dec, err := timePrecision(ctx, args[0])
+	if err != nil {
+		return nil, err
+	}
+	arg1Dec, err := timePrecision(ctx, args[1])
+	if err != nil {
+		return nil, err
+	}
+	bf.tp.Decimal = mathutil.Max(arg0Dec, arg1Dec)
```

Comment:
this is a bugfix?

Reply:
this decimal field is no use before this PR, so it got wrong value but not affect the user, but now if the duration comes from `chunk.row` will need it to compensate missed fsp info.